moveit_planners/trajopt/trajopt/src/collision_terms.cpp

```diff
@@ -0,0 +1,633 @@
+                                               const sco::VarVector& vars1)
+  : CollisionEvaluator(planning_scene, planning_group, safety_margin_data), m_vars0(vars0), m_vars1(vars1)
+{
+  // contact_manager_ = env_->getContinuousContactManager();
+  // contact_manager_->setActiveCollisionObjects(manip_->getLinkNames());
+  // contact_manager_->setContactDistanceThreshold(safety_margin_data_->getMaxSafetyMargin() +
+  //                                               0.04);  // The original implementation added a margin of 0.04;
+
+}
+
+// similar to CalcCollisions from SingleTimestepCollisionEvaluator except we need two states because it is the 
+// swept volume?
+void CastCollisionEvaluator::CalcCollisions(const DblVec& x, trajopt::ContactResultVector& dist_results)
+{
+  // tesseract::ContactResultMap contacts;
+  // tesseract::EnvStatePtr state0 = env_->getState(manip_->getJointNames(), sco::getVec(x, m_vars0));
+  // tesseract::EnvStatePtr state1 = env_->getState(manip_->getJointNames(), sco::getVec(x, m_vars1));
+  // ===> so far they create two states based on two sets of joint values
+
+  // for (const auto& link_name : manip_->getLinkNames())
+  //   contact_manager_->setCollisionObjectsTransform(link_name, state0->transforms[link_name], state1->transforms[link_name]);
+  // ===>>> in here, they add two collision transforms for each link
+
+  // ===>>> in the folloinwg, considering two collision transforms for each link, they calculate all the contacts among them
+  // contact_manager_->contactTest(contacts, tesseract::ContactTestTypes::ALL);
+  // tesseract::moveContactResultsMapToContactResultsVector(contacts, dist_results);
+
+  // In MoveIt we do have a checkCollision function that gets the two states in continous manner
+  // when using Bullet that function will be overriden by a real continous collision checking process
+  // All I have to do here is use that function for two 
+
+  // =====>>> So, I need to do it for two states as well.
+  // x has joint valus for all time steps: [j1_0 j2_0 ... j7_0, j1_1 j2_1 ... j7_1 ... j1_n j2_n ... j7_n]
+  Eigen::VectorXd joint_values_state_0 = sco::getVec(x, m_vars0);
+  std::vector<double> joint_values_DblVec_0 = std::vector<double>(joint_values_state_0.data(), joint_values_state_0.data() + joint_values_state_0.size());
+
+  moveit::core::RobotState& rob_state = planning_scene_->getCurrentStateNonConst();
+  const moveit::core::JointModelGroup* joint_model_group = rob_state.getJointModelGroup(planning_group_);
+
+  collision_detection::CollisionRequest collision_request;
+  collision_detection::CollisionResult collision_result;
+  collision_request.group_name = planning_group_;
+  // collision_request.distance = true;
+  collision_request.contacts = true;
+  collision_request.max_contacts = 100;
+  collision_request.max_contacts_per_pair = 5;
+  collision_request.verbose = false;
+
+  rob_state.setToDefaultValues();
```

Comment:
```suggestion
```
likely wrong if you have a gripper with another DOF attached to your robot but not part of the group

Reply:
I will remove `setToDefaultValues()`s as part of a cleanup commit